crates/sui-gateway/src/unit_tests/rpc_server_tests.rs

```diff
@@ -385,3 +386,130 @@ async fn test_get_fullnode_transaction() -> Result<(), anyhow::Error> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_get_fullnode_events() -> Result<(), anyhow::Error> {
```

Comment:
Like other large tests, I think this one would be easier to read & debug if we could split it into a test fixture with multiple cases: setup data before all tests, then do each assert as a separate test case in the fixture.